### PR TITLE
refactor: refactor error handling

### PR DESCRIPTION
**Pull request by tropicaldog**
## Overview

Refactor all instances where error handling is not properly handled

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved error handling and assignment order in block manager to ensure data consistency.
  - Enhanced error handling in test cases to ensure block height and initial height are verified correctly.

- **Tests**
  - Added additional error checks and assertions in test cases to improve test reliability and coverage.

### CODE REVIEW

File: block/manager_test.go [hunk 1]
```diff
@@ -64,6 +64,7 @@ func TestInitialStateClean(t *testing.T) {
 	es, _ := store.NewDefaultInMemoryKVStore()
 	emptyStore := store.New(es)
 	s, err := getInitialState(emptyStore, genesis)
+	require.NoError(err)
```
[the remark this answers is not in the document]
Comment (Manav-Aggarwal) [reply block/manager_test.go:67]: Yes instead of adding this line, the already existing line should be moved above.

File: block/manager_test.go [hunk 2]
```diff
@@ -90,6 +91,7 @@ func TestInitialStateStored(t *testing.T) {
 	err := store.UpdateState(ctx, sampleState)
 	require.NoError(err)
 	s, err := getInitialState(store, genesis)
+	require.NoError(err)
```
Comment (Manav-Aggarwal) [inline block/manager_test.go:94]: same as above

**Review by Manav-Aggarwal — CHANGES_REQUESTED** [review]
Left a small comment. Otherwise, LGTM

File: block/manager_test.go [hunk 3]
```diff
@@ -64,6 +64,7 @@ func TestInitialStateClean(t *testing.T) {
 	es, _ := store.NewDefaultInMemoryKVStore()
 	emptyStore := store.New(es)
 	s, err := getInitialState(emptyStore, genesis)
+	require.NoError(err)
 	require.Equal(s.LastBlockHeight, uint64(genesis.InitialHeight-1))
 	require.NoError(err)
```
Comment (Manav-Aggarwal) [inline block/manager_test.go:69]: This line should be removed since it's redundant

**Review by Manav-Aggarwal — CHANGES_REQUESTED** [review]

**Review by Manav-Aggarwal — APPROVED** [review]
Thank you @tropicaldog, LGTM

**Review by yarikbratashchuk — APPROVED** [review]
looks alright

**Pull request merged**